src/main/java/org/folio/rest/impl/TenantRefAPI.java

```diff
@@ -152,6 +153,18 @@ private void loadRef(Map<String, String> headers, Iterator<String> it, Handler<A
     }
   }
 
+  private void applyHeaders(HttpClientRequest req, Map<String, String> headers, String json) {
```

Comment:
This does not add all headers but only x headers.
And it invokes end.
I would have named it `endWithXHeaders`.